scripts/gen-req-full-stats.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  onboardingDataCollection,
+  semestersCollection,
+  toggleableRequirementChoicesCollection,
+  overriddenFulfillmentChoicesCollection,
+  courseFulfillmentStats,
+} from './firebase-config';
+
+import computeGroupedRequirementFulfillmentReports from '../src/requirements/requirement-frontend-computation';
+import computeFulfillmentStats from '../src/requirements/fulfillment-stats';
+import { createAppOnboardingData } from '../src/user-data-converter';
+
+import '../src/requirements/decorated-requirements.json';
+
+let idRequirementFrequency = new Map<string, Map<number, number>[]>();
+let counter = 0;
+/**
+ * Prints to console.out the json string of frequencies each course take to
+ * fulfill a given requirement
+ */
+async function computeRequirementFullfillmentStatistics(_callback) {
+  const semQuerySnapshot = await semestersCollection.get();
+  semQuerySnapshot.forEach(async doc => {
+    // obtain the user's semesters, onboarding data, etc...
+    const semestersAndPlans = await doc.data();
+    const onboardingData = (await onboardingDataCollection.doc(doc.id).get()).data();
+    const toggleableRequirementChoices =
+      (await toggleableRequirementChoicesCollection.doc(doc.id).get()).data() ?? {};
+    const overriddenFulfillmentChoices =
+      (await overriddenFulfillmentChoicesCollection.doc(doc.id).get()).data() ?? {};
+
+    if (
+      onboardingData !== undefined &&
+      semestersAndPlans.semesters !== undefined &&
+      toggleableRequirementChoices !== undefined &&
+      overriddenFulfillmentChoices !== undefined
+    ) {
+      // if the user has onboarding data
+      try {
+        const newOnboardingData = await createAppOnboardingData(onboardingData); // convert to new format
+        const res = await computeGroupedRequirementFulfillmentReports(
+          // compute fulfillment stats
+          semestersAndPlans.semesters,
+          newOnboardingData,
+          toggleableRequirementChoices,
+          overriddenFulfillmentChoices
+        );
+        idRequirementFrequency = await computeFulfillmentStats(
+          res.groupedRequirementFulfillmentReport,
+          idRequirementFrequency
+        ); // compute fulfillment stats
+      } catch {
+        console.log(counter, ' : Error in computing fulfillment stats for user: ', doc.id);
```

Comment:
Delete logging statements/replace with a comment. Instead of printing to console when errors occur, maybe we can set it up to send an email to us? Not sure if this is necessary if the vast majority of computations are fine

Reply:
I'd disagree with this; I think it would be bad for the script to silently fail